accounts/views.py

```diff
@@ -77,6 +77,71 @@
 def crash_me(request):
     raise Exception
 
+def login(request, template_name, authentication_form):
+    # Freesound-specific login view to check if a user has multiple accounts
+    # with the same email address. We can switch back to the regular django
+    # once all accounts are adapted
+    from django.contrib.auth import views as auth_views
+    response = auth_views.login(request, template_name=template_name, authentication_form=authentication_form)
+    if isinstance(response, HttpResponseRedirect):
+        # If there is a redirect it's because the login was successful
+        # Now we check is the logged in user has shared email problems
+        same_user_qs = request.user.profile.has_many_emails(return_qs=True)
+        if same_user_qs.exists():
+            # If the logged has an email shared with other accounts we figure out if we changed his email or we changed
+            # the emails from the other accounts which shared it
+            original_email = same_user_qs.first().main_orig_email
+            if request.user.email == original_email:
+                # If this account's email was not changed, simply continue with the login redirect (user won't notice)
+                return response
+            else:
+                # If we changed this account's email, we advise the user to chose a new email address
+                return HttpResponseRedirect(reverse("accounts-multi-email-cleanup")
+                                            + '?next=%s' % request.POST.get('next'))
+        else:
+            return response
+
+    return response
+
+
+@login_required
+def multi_email_cleanup(request):
+    # We make sure that the email problem still affects the user
+    # If the problem is not there anymore, then we can remove corresponding same_user objects
+    # and show a message telling user that email problems for the account were fixed
+
+    same_user_qs = request.user.profile.has_many_emails(return_qs=True)
+    if not same_user_qs.exists():
+        # User should have never come here
+        return HttpResponseRedirect(reverse('accounts-home'))
+
+    original_email = same_user_qs.first().main_orig_email
+    main_user = same_user_qs.first().main_user  # i.e. user that kept the original email
+    other_users_with_same_email = set()
+
+    if request.user == main_user:
+        # Main user should not be redirected here
+        return HttpResponseRedirect(reverse('accounts-home'))
+
+    # At this point, the current logged in user must be one of the users whose email
+    # was changed. We check if the user has changed his email again, and if he did we
+    # simply show a "your email problems have been fixed" message, otherwise we show
```

Comment:
I think the message is not really needed, I updated the comment: https://github.com/MTG/freesound/pull/763/commits/1133118c19bf09a07df6a6bded22d87d83d238b4